src/Tribe/Admin/Notice/Date_Based.php

```diff
@@ -0,0 +1,231 @@
+<?php
+/**
+ * Abstract for various date-based Marketing notices, e.g. Black Friday sales or special coupon initiatives.
+ *
+ * @since TBD
+ */
+
+namespace Tribe\Admin\Notice;
+
+use Tribe__Date_Utils as Dates;
+
+abstract class Date_Based {
+	/**
+	 * The slug used to make filters specific to an individual notice.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public $slug = '';
+
+	/**
+	 * Placeholder for start date string.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public $start_date;
+
+	/**
+	 * Placeholder for start time int.
+	 *
+	 * @since TBD
+	 *
+	 * @var int
+	 */
+	public $start_time;
+
+	/**
+	 * Placeholder for end date string.
+	 *
+	 * @since TBD
+	 *
+	 * @var string
+	 */
+	public $end_date;
+
+	/**
+	 * Placeholder for end time int.
+	 *
+	 * @since TBD
+	 *
+	 * @var int
+	 */
+	public $end_time;
+
+	/**
+	 * Whether or not The Events Calendar is active.
+	 *
+	 * @since TBD
+	 *
+	 * @var boolean
+	 */
+	public $tec_is_active;
+
+	/**
+	 * Whether or not Event Tickets is active.
+	 *
+	 * @since TBD
+	 *
+	 * @var boolean
+	 */
+	public $et_is_active;
+
+	public function __construct() {
+		$tribe_dependency    = tribe( \Tribe__Dependency::class );
+		$this->tec_is_active = $tribe_dependency->is_plugin_active( 'Tribe__Events__Main' );
+		$this->et_is_active  = $tribe_dependency->is_plugin_active( 'Tribe__Tickets__Main' );
+
+		$this->hook();
+	}
+
+	/**
+	 * Register the various Marketing notices.
+	 *
+	 * @since TBD
+	 */
+	public function hook() {
+		$this->hook_notice();
+	}
+
+	/**
+	 * Register the notice.
+	 *
+	 * @since TBD
+	 */
+	public function hook_notice() {
+		tribe_notice(
+			$this->slug,
+			[ $this, "display_notice" ],
+			[
+				'type'     => 'tribe-banner',
+				'dismiss'  => 1,
+				'priority' => -1,
+				'wrap'     => false,
+			],
+			[ $this, "should_display" ]
+		);
+	}
+
+	/**
+	 * HTML for the notice.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The HTML string to be displayed.
+	 */
+	abstract function display_notice();
+
+	/**
+	 * Whether the notice should display.
+	 *
+	 * @since TBD
+	 *
+	 * @return boolean $should_display Whether the notice should display or not.
+	 */
+	public function should_display() {
+		// If upsells have been manually hidden, respect that.
+		if ( defined( 'TRIBE_HIDE_UPSELL' ) && TRIBE_HIDE_UPSELL ) {
+			return false;
+		}
+
+		$now           = Dates::build_date_object( 'now', 'UTC' )->format( 'U' );
+		$notice_start = $this->get_start_time();
+		$notice_end   = $this->get_end_time();
```

Comment:
maybe push this stuff below the screen check so we're only doing things if necessary?